apps/designer/app/designer/shared/design-tokens-manager/hooks.ts

```diff
@@ -0,0 +1,14 @@
+import { useSubscribe } from "~/shared/pubsub";
+import { useDesignTokens } from "~/shared/nano-states";
+import type { DesignToken } from "@webstudio-is/design-tokens";
+
+declare module "~/shared/pubsub" {
```

Comment:
I would create a single d.ts file and extend the pubsub interface there, easy to view in a single place what pubsub supports. 
Same for example is doing svelte-kit in `app.d.ts`

(same for above `declare module`)

Reply:
We thought about it and went for a distributed approach, the idea is that each "feature" or even package declares it's events and yet they are all typechecked globally, so they can't override each other

Problem with having them centrally is 2:
1. it's going to be many over time
2. it's going to have dependencies to all possible modules to define types. It would turn into a god-module that depends on everything

Reply:
Alternatively we can make isolated pubsub for each feature like each nano-state

Reply:
Not sure how this could look like, but if you have an idea, let's create an issue

Reply:
@istarkov ?

Reply:
resolving this conversation for now, lets have a separate issue or a thread